hikari/auto_mod.py

```diff
@@ -0,0 +1,286 @@
+
+
+@attrs.define(kw_only=True, weakref_slot=False)
+class AutoModSendAlertMessage(PartialAutoModAction):
+    """Log the triggering content to a specific channel."""
+
+    channel_id: snowflakes.Snowflake = attrs.field()
+    """ID of the channel to log trigger events to."""
+
+
+@attrs.define(kw_only=True, weakref_slot=False)
+class AutoModTimeout(PartialAutoModAction):
+    """Timeout the triggering message's author for a specified duration.
+
+    This type can only be set for `KEYWORD` and `MENTION_SPAM` rules.
+
+    Requires the `MODERATE_MEMBERS` permission to use.
+    """
+
+    duration: datetime.timedelta = attrs.field()
+    """The total seconds to timeout the user for (max 2419200 seconds/4 weeks)."""
+
+
+@attrs.define(kw_only=True, weakref_slot=False)
+class AutoModBlockMemberAction(PartialAutoModAction):
+    """Prevents a member from using text, voice, or other interactions."""
+
+
+class AutoModEventType(int, enums.Enum):
+    """Type of event to check for an auto-moderation rule."""
+
+    MESSAGE_SEND = 1
+    """When a member sends or edits a message in the guild."""
+
+    MEMBER_UPDATE = 2
+    """When a member updates their guild or user profile."""
+
+
+class AutoModTriggerType(int, enums.Enum):
+    """Type of trigger for an auto-moderation rule."""
+
+    KEYWORD = 1
+    """Match message content against a list of keywords and regexs."""
+
+    SPAM = 3
+    """Discord's guild anti-spam system."""
+
+    KEYWORD_PRESET = 4
+    """Discord's preset keyword triggers."""
+
+    MENTION_SPAM = 5
+    """Match messages that exceed the allowed limit of role or user mentions."""
+
+    MEMBER_PROFILE = 6
+    """Match user profiles against a list of keywords and regexs."""
+
+
+class AutoModKeywordPresetType(int, enums.Enum):
+    """Discord's KEYWORD_PRESET type."""
+
+    PROFANITY = 1
+    """Trigger on words which may be considered forms of swearing or cursing."""
+
+    SEXUAL_CONTENT = 2
+    """Trigger on words that refer to explicit behaviour or activity."""
+
+    SLURS = 3
+    """Trigger on personal insults and words which "may be considered hate speech"."""
+
+
+@attrs_extensions.with_copy
+@attrs.define(kw_only=True, weakref_slot=False)
+class PartialAutoModTrigger:
+    """Base class representing the content a rule triggers on."""
+
+    type: AutoModTriggerType = attrs.field(eq=False, hash=False, repr=False)
+    """The type action this triggers."""
+
+
+@attrs.define(kw_only=True, weakref_slot=False)
+class KeywordTrigger(PartialAutoModTrigger):
+    """A trigger based on matching message content against a list of keywords."""
+
+    keyword_filter: typing.Sequence[str] = attrs.field(eq=False, hash=False, repr=False)
+    """The filter strings this trigger checks for.
+    This supports a wildcard matching strategy which is documented at
+    https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies.
+    """
```

Comment:
Also, mind checking if this renders properly and clickable in the docs? Im not sure if you might have to escape it

```suggestion

    This supports a wildcard matching strategy which is documented at
    https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies.
    """
```